src/cfn/index.ts

```diff
@@ -1178,8 +1178,13 @@ abstract class AbstractCloudFormationStackCommand {
   protected watchStackEvents: boolean = true;
 
   constructor(readonly argv: GenericCLIArguments, readonly stackArgs: StackArgs) {
-    // note, this.region is set in _setup after the cal to configureAWS
-    this.profile = this.argv.profile || this.stackArgs.Profile;// tslint:disable-line
+    // region, profile, and assumeRoleArn are the only used for cli output here
+    // configureAWS is called by loadStackArgs prior to this constructor
+    // TODO We should cleanup / dry-up the resolution rules for these.
+    this.region = def(getCurrentAWSRegion(), this.argv.region || this.stackArgs.Region);
+    this.profile = (            // the resolved profile
+      this.argv.profile || this.stackArgs.Profile
+        || process.env.AWS_PROFILE || process.env.AWS_DEFAULT_PROFILE); // tslint:disable-line
```

Comment:
Is there a way to get this off of `aws.config.credentials` (or similar)? It feels weird knowing about the priority of the profile resolution here

Reply:
It's not in there unfortunately. We have it here https://github.com/unbounce/iidy/blob/master/src/configureAWS.ts#L83. Maybe we could return the resolved config from `configureAWS` and thread it through. An alternative would be a global singleton we could set.

Reply:
Let's leave that for a future tweak.